katip/src/Katip/Scribes/Handle.hs

```diff
@@ -159,6 +174,27 @@ bracketFormat withColor verb Item {..} =
     renderSeverity' severity =
       colorBySeverity withColor severity (renderSeverity severity)
 
+-- getProcessTimes returns process wall time,
+-- but it is not clear how to convert ClockTick to time
+firstLogAt :: UTCTime
+{-# NOINLINE firstLogAt #-}
+firstLogAt = unsafePerformIO getCurrentTime
+
+-- | 'bracketFormat' produces prefix to verbose for an interative terminal.
+-- Fields such as host, day, process, thread, etc are not relevant
```

Comment:
I don't think I agree that this change should be made to the public library. I'm not seeing what it is about a terminal specifically where you wouldn't want to know the thread id or timestamp, for example. That seems like a specific use case. I agree that we should make it easy for users to specify their own format without a ton of internal knowledge about the library, though.

Reply:
Did you mean for this whole MR or just this function?

For context: This came out of a desire for simpler formatting in CLI applications that are used by generalists, not Haskellers and the process is not server-like and you don't care about central logging. I think I'd have to see what this looks like in practice to judge better (i.e. how the output looks). If there's some extra flexibility we should add to this API for a public release, that could work as well. Otherwise, we could keep this local to the use case.

Reply:
Mostly just responding to this function. I think the addition of `mkFileScribeWithFormatter` makes sense, since we can't break API for `mkFileScribe`. This gives us a way to construct a scribe with a given `ItemFormatter`. 

I'd probably also suggest dropping `termLogEnv`. If we're going to have a collection of many formatters to choose from, we should avoid creating matching log env initializers specialized to one format or we'll end up having to make an initializer for every format. 

I think we can keep adding `ItemFormatter`s and ideally in the haddocks give an example of what a line looks like. It might even be worth making a formatter that takes a configuration record with fields like `includeHost`, `includeThreadId`, etc so users can fine tune their formatter without having to write one totally from scratch.